WordPressComKit/SiteService.swift

```diff
@@ -1,44 +1,82 @@
 import Foundation
 import Alamofire
 
-public class SiteService {
-    public init() {}
-    
-    public func fetchSite(siteID: Int, completion: (Site?, NSError?) -> Void) {
-        Alamofire
-            .request(RequestRouter.Site(siteID: siteID))
-            .validate()
-            .responseJSON { response in
-                guard response.result.isSuccess else {
-                    completion(nil, response.result.error)
-                    return
-                }
-                
-                let json = response.result.value as? [String: AnyObject]
-                let site = self.mapJSONToSite(json!)
-                
-                completion(site, nil)
+extension Result {
+    func map<T>(transform: Value -> T) -> Result<T, Error> {
+        return flatMap({ .Success(transform($0)) })
+    }
+
+    func flatMap<T>(transform: Value -> Result<T, Error>) -> Result<T, Error> {
```

Comment:
So if I'm understanding the code correctly this transforms a `Result` if its a success with the passed in closure and if its a failure then it passes it back without transforming. I'm assuming there could be slight implications to memory since this is creating another Result struct but probably negligible impact.